R/make_dist_fns.R

```diff
@@ -272,6 +272,9 @@ make_p_fn <- function(ps, qs,
             result[inds] <- result[inds] + disc_ps[i] * disc_weight
         }
 
+        # ensure result is within interval [0, 1]
+        result <- pmin(pmax(result, 0), 1)
```

Comment:
test case resulted in a cdf value equal to something like 1 + 1e-16.  This is due to floating point issues.  Other tests check for (approximate) equality of cdf values with expected outputs, so, relying on those checks for approximate correctness, here I'm just forcing the return to be within the expected range.